client/trino-client/src/main/java/okhttp3/internal/tls/LegacyHostnameVerifier.java

```diff
@@ -0,0 +1,69 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package okhttp3.internal.tls;
+
+import javax.net.ssl.HostnameVerifier;
+import javax.net.ssl.SSLException;
+import javax.net.ssl.SSLSession;
+import javax.security.auth.x500.X500Principal;
+
+import java.security.cert.Certificate;
+import java.security.cert.X509Certificate;
+
+import static okhttp3.internal.Util.verifyAsIpAddress;
+import static okhttp3.internal.tls.OkHostnameVerifier.allSubjectAltNames;
+
+public class LegacyHostnameVerifier
+        implements HostnameVerifier
+{
+    public static final HostnameVerifier INSTANCE = new LegacyHostnameVerifier();
+
+    private LegacyHostnameVerifier() {}
+
+    @Override
+    public boolean verify(String host, SSLSession session)
+    {
+        if (OkHostnameVerifier.INSTANCE.verify(host, session)) {
+            return true;
+        }
+
+        // the CN cannot be used with IP addresses
+        if (verifyAsIpAddress(host)) {
+            return false;
+        }
+
+        // try to verify using the legacy CN rules
+        try {
+            Certificate[] certificates = session.getPeerCertificates();
+            X509Certificate certificate = (X509Certificate) certificates[0];
```

Comment:
can array be empty?

Reply:
This comes from `OkHostnameVerifier`, so if it is, we would crash anyway, as we delegate to that code first. Based on the Javadoc, this doesn't seem possible:

> an ordered array of peer certificates, with the peer's own certificate first followed by any certificate authorities